holmes/plugins/toolsets/opensearch/opensearch_logs.py

```diff
@@ -140,8 +140,8 @@ def _get_fields_using_mappings(
     ) -> StructuredToolResult:
         """Use the OpenSearch getMappings API to retrieve fields (new implementation)"""
         url = urljoin(
-            self._toolset.opensearch_config.opensearch_url,
-            f"/{self._toolset.opensearch_config.index_pattern}/_mapping",
+            self._toolset.typed_config.opensearch_url,  # type: ignore
+            f"/{self._toolset.typed_config.index_pattern}/_mapping",  # type: ignore
```

Comment:
Why are `type: ignore` added here in multiple places? Typing is here as a safety mechanism against bugs that we should not ignore.

Would adding a not None check at the beginning of the invoke method help?

```
            if not self._toolset.typed_config:
                return StructuredToolResult(
                    status=ToolResultStatus.ERROR,
                    error=TOOLSET_CONFIG_MISSING_ERROR,
                    params=params
                )
```